### PR TITLE
Add jdbc connector e2e test

### PR DESCRIPTION
**Pull request by zhangyuge1**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by CalvinKirs** [comment]
#2255 . License it's ok.  I will take a look when my free.

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/FakeSourceToJdbcIT.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        // query result
+        String sql = "select * from test";
+        try (Connection connection = DriverManager.getConnection(psl.getJdbcUrl(), psl.getUsername(), psl.getPassword())) {
+            Statement statement = connection.createStatement();
+            ResultSet resultSet = statement.executeQuery(sql);
+            List<String> result = Lists.newArrayList();
+            while (resultSet.next()) {
+                result.add(resultSet.getString("name"));
+            }
+            Assertions.assertFalse(result.isEmpty());
```
Comment (ic4y) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/FakeSourceToJdbcIT.java:87]: Only check is not isEmpty that is not accurate.

**Comment by ic4y** [comment]
@zhangyuge1 Thank you for your contribution.

At present, only batch tests are covered, Streaming Mode and Exactly-Once Streaming Mode are not covered。Can you add these two test scenarios? And abstract public methods for use by mysql, pgsql, and dmsql . . .

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/FakeSourceToJdbcIT.java (hunk 1 above)
Comment (ruanwenjun) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/FakeSourceToJdbcIT.java:87]: Yes, we need to do data quality check.

**Review by ruanwenjun — APPROVED** [review]
LGTM, this PR only test the jdbc connector can work in batch mode, but don't check the data quality, this can be implemented in another PR.

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**